packages/block-editor/src/utils/dom.js

```diff
@@ -79,16 +89,22 @@ export function hasInnerBlocksContext( element ) {
 /**
  * Finds the block client ID given any DOM node inside the block.
  *
- * @param {Node} node DOM node.
+ * @param {Node?} node DOM node.
  *
- * @return {string|undefined} Client ID or undefined if the node is not part of a block.
+ * @return {string|undefined} Client ID or undefined if the node is not part of
+ *                            a block.
  */
 export function getBlockClientId( node ) {
-	if ( node.nodeType !== node.ELEMENT_NODE ) {
-		node = node.parentElement;
+	while ( node && node.nodeType !== window.Node.ELEMENT_NODE ) {
+		node = node.parentNode;
+	}
+
+	if ( ! node ) {
+		return;
 	}
 
-	const blockNode = node.closest( '.block-editor-block-list__block' );
+	const elementNode = /** @type {Element} */ ( node );
+	const blockNode = elementNode.closest( '.block-editor-block-list__block' );
```

Comment:
Oh, I guess I should note that I changed the implementation here, because the addition of type-checking caught some legitimate problems with the current implementation:

1. `node.parentElement` can be `null`.
2. `node.parentElement` may not exist depending on the node type (see https://github.com/WordPress/gutenberg/issues/21276#issuecomment-607588622)